ballerina/tests/consume_constraint_tests.bal

```diff
@@ -0,0 +1,322 @@
+    check channelListener.'start();
+    runtime:sleep(2);
+    test:assertEquals(receivedIntMaxValueConstraintError, "Validation failed for 'maxValue' constraint(s).");
+    check channelListener.gracefulStop();
+}
+
+@test:Config {enable: true}
+function numberMaxValueConstraintListenerPayloadTest() returns error? {
+    Service numberConstraintService =
+    @ServiceConfig {
+        queueName: CONSTRAINT_NUMBER_MAX_VALUE_QUEUE
+    }
+    service object {
+        remote function onMessage(Weight message) returns error? {
+            log:printInfo("The message received: " + message.toString());
+        }
+
+        remote function onError(Message msg, Error e) {
+            if e is PayloadValidationError {
+                receivedNumberMaxValueConstraintError = e.message();
+            }
+            log:printInfo("An error received in onError: " + e.message());
+        }
+    };
+    check produceMessage(1099.999.toString(), CONSTRAINT_NUMBER_MAX_VALUE_QUEUE);
+    Listener channelListener = check new (DEFAULT_HOST, DEFAULT_PORT);
+    check channelListener.attach(numberConstraintService);
+    check channelListener.'start();
+    runtime:sleep(2);
+    test:assertEquals(receivedNumberMaxValueConstraintError, "Validation failed for 'maxValue' constraint(s).");
+    check channelListener.gracefulStop();
+}
+
+@test:Config {enable: true}
+function numberMinValueConstraintListenerPayloadTest() returns error? {
+    Service numberConstraintService =
+    @ServiceConfig {
+        queueName: CONSTRAINT_NUMBER_MIN_VALUE_QUEUE
+    }
+    service object {
+        remote function onMessage(Weight message) returns error? {
+            log:printInfo("The message received: " + message.toString());
+        }
+
+        remote function onError(Message msg, Error e) {
+            if e is PayloadValidationError {
+                receivedNumberMinValueConstraintError = e.message();
+            }
+            log:printInfo("An error received in onError: " + e.message());
+        }
+    };
+    check produceMessage(1.999.toString(), CONSTRAINT_NUMBER_MIN_VALUE_QUEUE);
+    Listener channelListener = check new (DEFAULT_HOST, DEFAULT_PORT);
+    check channelListener.attach(numberConstraintService);
+    check channelListener.'start();
+    runtime:sleep(2);
+    test:assertEquals(receivedNumberMinValueConstraintError, "Validation failed for 'minValue' constraint(s).");
+    check channelListener.gracefulStop();
+}
+
+@test:Config {}
+function recordConstraintMessageValidTest() returns error? {
+    Child child = {name: "PhilDunphy", age: 27};
+    check produceMessage(child.toString(), CONSTRAINT_VALID_RECORD_QUEUE);
+    Client 'client = check new (DEFAULT_HOST, DEFAULT_PORT);
+    Child|error result = 'client->consumePayload(CONSTRAINT_VALID_RECORD_QUEUE);
+    if result is error {
+        test:assertFail(result.message());
+    } else {
+        test:assertEquals(result, child);
+    }
+    check 'client->close();
+}
```

Comment:
Do we have any test case by disabling the config?

Reply:
Added in https://github.com/ballerina-platform/module-ballerinax-rabbitmq/pull/470/commits/7342c94795ad8287bc499803c68ecd8e27849a0b